src/parser/parsers/changelogs/fetch_changelogs.py

```diff
@@ -0,0 +1,67 @@
+import os
+from os import listdir
+from os.path import isfile, join
+import feedparser
+from bs4 import BeautifulSoup
+from urllib import request
+
+
+class ChangelogFetcher:
+    def __init__(
+        self,
+        output_dir,
+        txt_path=os.path.join(os.path.dirname(__file__), '../../raw-changelogs/'),
+        rss_feed='https://forums.playdeadlock.com/forums/changelog.10/index.rss',
+    ):
+        self.CHANGELOGS_DIR = txt_path
+        self.RSS_URL = rss_feed
+        self.OUTPUT_CHANGELOGS = output_dir + '/raw-changelogs'
+        self.changelogs_by_date = {}
+
+    def fetch_update_html(self, link):
+        html = request.urlopen(link).read()
+        soup = BeautifulSoup(html, features='html.parser')
+        entries = []
+        # Find all <div> tags with class 'bbWrapper' (xenforo message body div)
+        for div in soup.find_all('div', class_='bbWrapper'):
+            entries.append(div.text.strip())
+        return entries
+
+    # download rss feed from changelog forum and parse entries
+    def fetch_forum_changelogs(self):
+        print('Parsing Changelog RSS feed')
+        # fetches 20 most recent entries
+        feed = feedparser.parse(self.RSS_URL)
+        out_dir = self.OUTPUT_CHANGELOGS + '/rss'
```

Comment:
I think we just need one directory for raw changelogs (current raw-changelogs dir), not three. When the parser runs, it will:
1. Query rss feed to get the latest changelog dates
2. See if any are missing in raw-changelogs
3. If they are missing, query for them and writes to the raw-changelogs dir

Reply:
I ended up created an `input-data` directory, and placed the raw changelogs in there.  I noticed while running it there are some syntax and spelling fixes in the changelogs, which I tried to preserve.  It works similar to what you outlined:
1. queries rss, get's latest posts, and fetches each posts content and stores is keyed on date.  
2. goes through each file in raw-changelogs, and adds / replaces the changelog if it already existed. (good fixing changelog text for reasons)
3. takes combined dict of changelogs, write to output files
4. passes changeslogs to to the parser, which extracts out the final specific changelog files.

Reply:
Ah good idea with the input data, didn't consider fixing changelogs at this point, but that's easier than once its parsed.

I'd say all you need is a check for whether it exists in output-data already, as querying every time has no use if its already been done once.

I can see you've added a `CHLOGS` flag to maybe mitigate this, but the same issue will come up when you are trying to develop on the changelog parser. So I'm not sure I see a need for this flag, happy to be proven wrong though

Reply:
Yeah, I expanded it out to be "CHANGELOGS", and got rid of the SKIP_RSS flag
Now, It'll fetch the rss like usual but if a changelogs entry already exists, it won't re-fetch them.  this is done by loading in changelogs from the output directory first, then the rss feed, then the txt input directory.